cmd/poller/poller.go

```diff
@@ -754,21 +756,12 @@ func init() {
 	_ = pollerCmd.MarkFlagRequired("poller")
 }
 
-// start poller, if fails try to write to syslog
+// start poller, if fails write to log
 func main() {
 
-	// don't recover if a goroutine has panicked, instead
-	// try to zeroLog as much as possible, since normally it's
-	// not properly logged
 	defer func() {
-		//logger.Warn("(main) ", "defer func here")
 		if r := recover(); r != nil {
-			logger.Info().Msgf("harvest poller panicked: %v", r)
-			// if logger still available try to write there as well
-			// do this last, since might make us panic as again
-			logger.Fatal().Msgf("(main) %v", r)
-			logger.Fatal().Msg(`(main) terminating abnormally, tip: run in foreground mode (with "--loglevel 0") to debug`)
-
+			logger.Fatal().Stack().Err(errors.New(errors.GO_ROUTINE_PANIC, string(debug.Stack()))).Msg(`(main) terminating abnormally, tip: run in foreground mode (with "--loglevel 0") to debug`)
```

Comment:
Same comment here

Reply:
recover return type is not error. One way to get stacktrace is like this.